hudi-io/src/main/java/org/apache/hudi/io/hfile/HFileBlockType.java

```diff
@@ -0,0 +1,186 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.hudi.io.hfile;
+
+import java.io.DataInputStream;
+import java.io.IOException;
+
+import static java.nio.charset.StandardCharsets.UTF_8;
+import static org.apache.hudi.io.hfile.DataSize.MAGIC_LENGTH;
+
+/**
+ * Represents the HFile block type.
+ */
+public enum HFileBlockType {
+  /**
+   * Data block, both versions
+   */
+  DATA("DATABLK*", BlockCategory.DATA),
+
+  /**
+   * An encoded data block (e.g. with prefix compression), version 2
+   */
+  ENCODED_DATA("DATABLKE", BlockCategory.DATA) {
+    @Override
+    public int getId() {
+      return DATA.ordinal();
+    }
+  },
+
+  /**
+   * Version 2 leaf index block. Appears in the data block section
+   */
+  LEAF_INDEX("IDXLEAF2", BlockCategory.INDEX),
+
+  /**
+   * Bloom filter block, version 2
+   */
+  BLOOM_CHUNK("BLMFBLK2", BlockCategory.BLOOM),
```

Comment:
Did you get around to reading these? Writing and reading bloomfilters

Reply:
JIRA to track: https://issues.apache.org/jira/browse/HUDI-7216